### ISSUE TITLE
[JENKINS-20967] Cloud provisioning called when Jenkins is quieting Down

### ISSUE THREAD

**Issue by jenkins-infra-bot** [timja-org/jenkins-gh-issues-poc-11-23#3874]
<p>If Jenkins is quieting down and there are builds in the queue, nodes are still provisioned from any clouds.</p>

<p>Ideally, Jenkins would not provision new slaves when it is supposed to be quieting down.</p>

---
<details><summary><i>Originally reported by <img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/recampbell.png" title="recampbell's avatar" /> <a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=recampbell">recampbell</a>, imported from: <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-20967" target="_blank">Cloud provisioning called when Jenkins is quieting Down</a></i></summary>
<i><ul>
<li><b>status</b>: Open
<li><b>priority</b>: Major
<li><b>component(s)</b>: core
<li><b>resolution</b>: Unresolved
<li><b>votes</b>: 2
<li><b>watchers</b>: 6
<li><b>imported</b>: 2025-11-23
</ul></i>
<details><summary>Raw content of original issue</summary>

<pre>
<p>If Jenkins is quieting down and there are builds in the queue, nodes are still provisioned from any clouds.</p>

<p>Ideally, Jenkins would not provision new slaves when it is supposed to be quieting down.</p></pre>
</details>
</details>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/recampbell.png" title="recampbell's avatar" /> recampbell:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-20967?focusedId=191176&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-191176">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>It seems that the root cause is that each label's LoadStatistics are still reporting queue lengths over zero.</p>

<p>So a simple fix would just have hudson.slaves.NodeProvisioner#update not provision if Jenkins.isQuietingDown().</p>
</pre>
</details>

</ul>
</details>

<p>It seems that the root cause is that each label's LoadStatistics are still reporting queue lengths over zero.</p>

<p>So a simple fix would just have hudson.slaves.NodeProvisioner#update not provision if Jenkins.isQuietingDown().</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/thomassuckow.jpg" title="thomassuckow's avatar" /> thomassuckow:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-20967?focusedId=222499&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-222499">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>Changing NodeProvisioner would create a deadlock situation with NonBlockingTasks, such as a Matrix Build. Their slaves may never get created. I think it would be more appropriate to modify the behaviour of countBuildable*() in Queue to only count tasks that are not blocked by shutdown.</p>

<p>I have another pull request manipulating countBuildable, I may make a pull request for this after that one gets accepted.</p>
</pre>
</details>

</ul>
</details>

<p>Changing NodeProvisioner would create a deadlock situation with NonBlockingTasks, such as a Matrix Build. Their slaves may never get created. I think it would be more appropriate to modify the behaviour of countBuildable*() in Queue to only count tasks that are not blocked by shutdown.</p>

<p>I have another pull request manipulating countBuildable, I may make a pull request for this after that one gets accepted.</p>
